docs/locks.rst

```diff
@@ -3,5 +3,73 @@
 
 .. versionadded:: 4.2
 
+Coordinate coroutines with synchronization primitives analogous to those the
+standard library provides to threads.
+
+*(Note that these primitives are not actually thread-safe and cannot be used in
+place of those from the standard library--they are meant to coordinate Tornado
+coroutines in a single-threaded app, not to protect shared objects in a
+multithreaded app.)*
+
+.. _the-wait-notify-pattern:
+
+The Wait / Notify Pattern
+=========================
+Tornado locks follow a "wait / notify pattern": one coroutine waits to be
+notified by another. Take `~tornado.locks.Condition` as an example:
+
+.. testcode::
+
+    from tornado import ioloop, gen, locks
+
+
+    loop = ioloop.IOLoop.current()
```

Comment:
Nit: We currently use `io_loop` fairly consistently as an IOLoop variable in the docs.